docs/conceptual/core/storage.md

```diff
@@ -2,8 +2,87 @@
 title: Storage
 ---
 
-* Key Value Database, RocksDB
+The Substrate uses a simple a key-value data store implemented as a
+database-backed modified Merkle tree.
 
-* Trie abstraction
+## Key-Value Database
 
-* Runtime Storage abstraction
+Substrate implements its storage database with [RocksDB](https://rocksdb.org/),
+a persistent key-value store for fast storage environments.
+
+This is used for all the components of Substrate which require persistent
+storage such as:
+
+- Substrate clients
+- Substrate light-clients
+- Off-chain workers
+
+## Trie Abstraction
+
+One advantage of using a simple key-value store is that you are able to easily
+abstract other storage structures on top.
+
+Substrate uses a Base-16 Modified Merkle Patricia tree ("trie") from
+[`paritytech/trie`](https://github.com/paritytech/trie) to provide a trie
+structure whose contents can be modified and whose root hash is recalculated
+efficiently.
+
+Tries are important tool for blockchains because they allow for efficient
+storing and sharing of the historical block state. The root hash of the trie
+provides a cryptographic fingerprint of the final resulting storage after
+executing all state transitions by a blockchain. Thus, two blockchain nodes can
+easily verify they have the same final state by simply comparing their trie
+root.
+
+Accessing trie data is costly. Each read operation takes O(log N) time, where N
+is the number of elements stored in the trie. To mitigate this, we use a key
+value cache.
+
+All trie nodes are stored in RocksDB and part of the trie state can get pruned,
+i.e. a key-value pair can be deleted from the storage when it is out of pruning
+range for non archive nodes. The trie nodes partial path prefixes the hash of
```

Comment:
"The trie nodes partial path prefixes"

Super confusing. You were just talking about nodes as nodes in a network. Now, nodes means nodes in the trie.

Something like, "The partial path of a node in the trie is the prefix for..." may be easier to digest.

Reply:
I just removed this sentence. I think the value add is low. feels like an implementation detail.